documentation/minimal_viable_product_description.md

```diff
@@ -0,0 +1,125 @@
+
+---
+
+## **Long-Term Considerations (Post-MVP)**
+
+The post-MVP roadmap includes the following high-priority goals:
+
+- Addition of uncertainty estimation (approximate posterior distributions).
+- Expansion of prediction targets to include additional measures of violence and conflict impacts.
+- Integration of nowcasting and input/interpolation uncertainty.
+- Implementation of a unified evaluation framework, followed by the rollout of an online evaluation system.
+- Global expansion of PGM-level forecasts.
+- Expansion of the evaluation metric roster to included all metrics decided on doing the "metric workshop"
+
+----
+
+## **Suggested formulation of GitHub Milestones (documented here for the purpose of PR review)**
+
+### **1. PGM Predictions (Legacy Models)**
+
+#### **Description**  
+Produce PGM-level predictions using the majority of models from the old pipeline, focusing on consistency with the original pipeline's output.
+
+#### **Subtasks**  
+   - Migrate and implement simple stepshift models first.
+   - Follow up with the implementation of hurdle models.
+   
+#### **Outcome**  
+PGM models from the old pipeline are successfully migrated and integrated into the new pipeline, producing predictions that are consistent with the original outputs.
+
+
+### **2. HydraNet PGM Predictions**
+
+#### **Description**  
+Integrate HydraNet models to produce PGM-level predictions within the new pipeline, ensuring their full deployment and operational status.
+
+#### **Outcome**  
+At least one HydraNet model is fully integrated into the new pipeline, generating accurate PGM-level predictions.
+
+### **3. CM Predictions (Legacy Models)**
+
+#### **Description**  
+Migrate CM-level models from the old pipeline to the new pipeline, aiming for consistent predictions with the original pipeline's outputs.
+
+#### **Subtasks**  
+   - Migrate and implement simple stepshift models first.
+   - Follow up with the implementation of hurdle models.
+   
+#### **Outcome**  
+CM models from the old pipeline are successfully migrated and produce predictions consistent with the original pipeline.
+
+### **4. Ensembles (Mean and Median)**
+
+#### **Description**  
+Implement a mean ensemble method for both PGM and CM-level models. Additionally, deploy a median ensemble as a shadow model to provide a baseline comparison.
+
+#### **Outcome**  
+Mean and median ensemble methods are implemented, with evaluation results documented and reported on Weights and Biases.
+
+### **5. Calibration (Temporary Solution)**
+
+#### **Description**  
+Use Jim’s calibration script to temporarily align ensemble predictions at the PGM and CM levels.
+
+#### **Outcome**  
+A temporary calibration solution is in place, aligning PGM and CM ensemble predictions, with improvements planned for post-MVP development.
+
```

Comment:
Note the within-level calibration noted above. It should be done at the constituent model level - it makes more sense to have an unweighted average of well-calibrated models than of poorly-calibrated ones. If some are poorly calibrated they will pull the entire ensemble down.

Reply:
Yes, my intention is to separately calibrate all the ppm models before they are ensembles (i.e. averaged, at this stage)

Regarding suggesting imputation as a deliverable to the FAO, on what timescale will they expect this to be delivered? We have not discussed imputation itself for some time, unless we are lumping imputation in with input-uncertainty.